wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -810,8 +810,15 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
             Context.propagate(walletApplication.wallet.context)
             var encryptionKey: KeyParameter? = null
 
-            var lastContactRequestTime = if (dashPayContactRequestDao.countAllRequests() > 0)
-                dashPayContactRequestDao.getLastTimestamp() - DateUtils.MINUTE_IN_MILLIS * 12
+            var lastContactRequestTime = if (dashPayContactRequestDao.countAllRequests() > 0) {
+                val lastTimeStamp = dashPayContactRequestDao.getLastTimestamp()
+                // if the last contact request was received in the past 10 minutes, then query for
+                // contact requests that are 10 minutes before it.  If the last contact request was
+                // more than 10 minutes ago, then query all contact requests that came after it.
+                if (lastTimeStamp < System.currentTimeMillis() - DateUtils.MINUTE_IN_MILLIS * 10)
```

Comment:
I would prefer to use `TimeUnit.MINUTES.toMillis(10)` instead of `DateUtils.MINUTE_IN_MILLIS * 10` but it isn't a part of this story so we can let well alone ;)

Reply:
Looks like our app is using both of these methods depending on the author.